src/main/java/org/jenkinsci/plugins/durabletask/BourneShellScript.java

```diff
@@ -191,23 +184,24 @@ public String getScript() {
         args.addAll(Arrays.asList("sh", "-c", cmd));
         LOGGER.log(Level.FINE, "launching {0}", args);
         Launcher.ProcStarter ps = launcher.launch().cmds(args).envs(escape(envVars)).pwd(ws).quiet(true);
-        boolean novel;
-        synchronized (encounteredPaths) {
-            Integer cnt = encounteredPaths.get(ws);
-            if (cnt == null) {
-                cnt = 0;
-            }
-            novel = cnt < NOVEL_WORKSPACE_DIAGNOSTICS_COUNT;
-            encounteredPaths.put(ws, cnt + 1);
-        }
-        if (novel) {
-            // First time in this combination. Display any output from the wrapper script for diagnosis.
+        if (LAUNCH_DIAGNOSTICS) {
             ps.stdout(listener);
         } else {
-            // Second or subsequent time. Suppress output to save a thread.
             ps.readStdout().readStderr(); // TODO RemoteLauncher.launch fails to check ps.stdout == NULL_OUTPUT_STREAM, so it creates a useless thread even if you never called stdout(…)
         }
-        ps.start();
+        final Proc proc = ps.start();
+        if (!LAUNCH_DIAGNOSTICS && proc instanceof AutoCloseable) {
+            Timer.get().schedule(new Runnable() {
+                @Override public void run() {
+                    try {
+                        LOGGER.log(Level.FINE, "cleaning up {0} in {1}", new Object[] {proc, controlDir});
+                        ((AutoCloseable) proc).close();
+                    } catch (Exception x) {
+                        LOGGER.log(Level.WARNING, "failed to clean up " + proc + " in " + controlDir, x);
+                    }
+                }
+            }, 1, TimeUnit.MINUTES);
```

Comment:
I'm not sure I understand how long the process is supposed to live.

Reply:
Hours perhaps. `ContainerExecProc.close` merely stops listening to it, because it should have no stdio and an exit status of 0—this is the wrapper process. The stdio and exit status of the user process are captured in files, so this `Proc` handle is worthless (note that we throw it away before returning from the method).